### PR TITLE
[CUR-427] feat: tier / child subject downloads

### PR DESCRIPTION
**Pull request by sonalisin**
# How to review this PR

_Leave this text block for the reviewer_

- Check [component hierarchy](https://miro.com/app/board/uXjVNnKBgyk=/?share_link_id=59445593794) is followed correctly
- Check the design [Heuristics](https://components.thenational.academy/?path=/docs/docs-howtodesigncomponents--docs#heuristics-for-component-design) have been followed
- Check [naming conventions](https://components.thenational.academy/?path=/docs/docs-namingconventions--docs) have been applied
- Check for these gotchyas:
  - Missing exports for Oak components
  - Accidental export of Internal components
  - Snapshots of unexpected components have been modified
  - Circular dependencies
  - Code duplication (via not using base components)
  - Non-functional storybook for this or related components
  - Sensitve files changed eg. atoms, or style tokens
  - Relative imports
  - Default exports

# Add your PR description below

- Created an organism for the subject / tier selector 
- _Caveat: the default values are set in the component, but I'm not sure why Storybook doesn't always display them_ : I removed the additional story only displaying tier and this fixed it. I'm not sure if this is a Stories thing...

## Link to the design doc
https://www.figma.com/design/3cz9fb0ObFDLv1c212OtJv/Final-designs---Curriculum-squad?node-id=3984-20165&m=dev

## A link to the component in the deployment preview
https://deploy-preview-178--lively-meringue-8ebd43.netlify.app/?path=/docs/components-organisms-curriculum-oakdownloadsjourneychildsubjecttierselector--docs

## Testing instructions
- Select different combinations, and open the console
- When the user presses submit, the selected subject / tier slug should be displayed in the console

### CODE REVIEW

File: src/components/organisms/curriculum/OakDownloadsJourneyChildSubjectTierSelector/OakDownloadsJourneyChildSubjectTierSelector.tsx [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+  OakRadioGroup,
+} from "@/components/molecules";
+
+// const StyledOakFlex = styled(OakFlex)`
+//   width: ${parseSpacing("all-spacing-6")};
+// `;
+// For example you could restyle the OakFlex component by adding the styles to the css template literal below
+export interface Tier {
+  tier: string;
+  tier_slug: string;
```
Comment (kimon-satan) [inline src/components/organisms/curriculum/OakDownloadsJourneyChildSubjectTierSelector/OakDownloadsJourneyChildSubjectTierSelector.tsx:19]: it's interesting that you guys are working with snake_case here. My view is that it should be snake_case at DB level and then we should convert immediatetly and consistently to camelCase for React world. We've been doing this already on our queries in OWA>

File: src/components/organisms/curriculum/OakDownloadsJourneyChildSubjectTierSelector/OakDownloadsJourneyChildSubjectTierSelector.tsx [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+    setTierSelected(tierSlug);
+  }
+
+  function handleNextClick() {
+    getTierSubjectValues(tierSelected, childSubjectSelected);
+  }
+
+  return (
+    <OakBox>
+      <OakHeading $font={"heading-4"} tag="h4" $mb={"space-between-m"}>
```
Comment (kimon-satan) [inline src/components/organisms/curriculum/OakDownloadsJourneyChildSubjectTierSelector/OakDownloadsJourneyChildSubjectTierSelector.tsx:85]: I'm personally anti using margin whenever possible. It breaks component encapsulation in a subtle way and leads to layout issues down the line. A combination of Flexbox gap and padding normally does the same job.

File: src/components/organisms/curriculum/OakDownloadsJourneyChildSubjectTierSelector/OakDownloadsJourneyChildSubjectTierSelector.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,41 @@
+import React from "react";
+import { StoryObj, Meta } from "@storybook/react";
+
+import { OakDownloadsJourneyChildSubjectTierSelector } from "./OakDownloadsJourneyChildSubjectTierSelector";
+
+const meta: Meta<typeof OakDownloadsJourneyChildSubjectTierSelector> = {
```
Comment (kimon-satan) [inline src/components/organisms/curriculum/OakDownloadsJourneyChildSubjectTierSelector/OakDownloadsJourneyChildSubjectTierSelector.stories.tsx:6]: It's quite good to define the argTypes and controls here to make the story more useable for others.

**Review by kimon-satan — APPROVED** [review]
This looks pretty decent to me. I've left a bunch of minor style points and some just tidy up and some more matters of opinion. The snake_case camelCase one would be worth discussing on cross-sq channel

File: src/components/organisms/curriculum/OakDownloadsJourneyChildSubjectTierSelector/OakDownloadsJourneyChildSubjectTierSelector.tsx (hunk 1 above)
Comment (sonalisin) [reply src/components/organisms/curriculum/OakDownloadsJourneyChildSubjectTierSelector/OakDownloadsJourneyChildSubjectTierSelector.tsx:19]: Good point, it makes sense to update this to camel case for front end, I just hadn't noticed when copying the types across.. Will update in the component and make some tickets in OWA for consistency

**Review by rightsaidjames — APPROVED** [review]
Overall this looks good, but if possible it would be good to be able to preview in Storybook what happens when only tiers are available for selection. Certainly not a blocker for merge, though.

**Comment by orangemug** [comment]
In the storybook preview the focus indicator is outside the button, see

Screenshot 2024-05-15 at 15 12 32
Screenshot 2024-05-15 at 15 12 52

**Pull request merged**